spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/ReactiveRedisStreamMessageHandler.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Implementation of {@link org.springframework.messaging.ReactiveMessageHandler} which writes
+ * Message payload into a Redis stream , using reactive stream operation.
+ *
+ * @author Attoumane AHAMADI
+ *
+ * @since 5.3
+ */
+public class ReactiveRedisStreamMessageHandler extends AbstractReactiveMessageHandler {
+
+	private final Expression streamKeyExpression;
+
+	private volatile EvaluationContext evaluationContext;
+
+	private boolean extractPayload = true;
+
+	private ReactiveStreamOperations reactiveStreamOperations;
+
+	private String streamKey;
+
+	private RedisSerializationContext serializationContext = RedisSerializationContext.string();
+
+	private ReactiveRedisConnectionFactory connectionFactory;
+
+	public ReactiveRedisStreamMessageHandler(Expression streamKeyExpression,
+			ReactiveRedisConnectionFactory connectionFactory) {
+		Assert.notNull(streamKeyExpression, "'streamKeyExpression' must not be null");
+		Assert.notNull(connectionFactory, "'connectionFactory' must not be null");
+		this.streamKeyExpression = streamKeyExpression;
+		this.connectionFactory = connectionFactory;
+	}
+
+	public ReactiveRedisStreamMessageHandler(String streamKey, ReactiveRedisConnectionFactory connectionFactory) {
+		this.setStreamKey(streamKey);
+		Assert.notNull(connectionFactory, "'connectionFactory' must not be null");
+		this.connectionFactory = connectionFactory;
+		this.streamKeyExpression = new LiteralExpression(streamKey);
+	}
+
+	public void setStreamKey(String streamKey) {
+		Assert.hasText(streamKey, "'streamKey' must not be an empty string.");
+		this.streamKey = streamKey;
+	}
+
+	public void setSerializationContext(RedisSerializationContext serializationContext) {
+		Assert.notNull(serializationContext, "'serializationContext' must not be null");
+		this.serializationContext = serializationContext;
+	}
+
+	public void setExtractPayload(boolean extractPayload) {
+		this.extractPayload = extractPayload;
+	}
+
+	void setEvaluationContext(EvaluationContext evaluationContext) {
+		Assert.notNull(evaluationContext, "'evaluationContext' must not be null");
+		this.evaluationContext = evaluationContext;
+	}
+
+	@Override
+	public String getComponentType() {
+		return "redis:stream-outbound-channel-adapter";
+	}
+
+	@Override
+	protected Mono<Void> handleMessageInternal(Message<?> message) {
+		initStreamOperations();
+
+		if (!StringUtils.hasText(this.streamKey)) {
+			this.streamKey = this.streamKeyExpression.getValue(this.evaluationContext, message, String.class);
+		}
+
+		Object value = message;
+		if (this.extractPayload) {
+			value = message.getPayload();
+		}
+
+		ObjectRecord<String, Object> record = StreamRecords
+				.<String, Object>objectBacked(value)
+				.withStreamKey(this.streamKey);
+
+		return this.reactiveStreamOperations.add(record);
```

Comment:
We probably also need to think about a batch insert support.
see this one:
```
default Flux<RecordId> add(K key, Publisher<? extends Map<? extends HK, ? extends HV>> bodyPublisher) {
```

Reply:
How to handle that case? Do we need a `batchMode` boolean? I've rarely had the opportunity to use the `Publisher` interface.

Reply:
No problem! Let's consider this as a *feature request* when we are already done with the basics we have here!